app/webpacker/components/ContactsPage/ContactForm.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useMemo, useRef, useState } from 'react';
+import {
+  Form, FormGroup, FormField, Button, Radio, Message,
+} from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { contactUrl } from '../../lib/requests/routes.js.erb';
+import useInputState from '../../lib/hooks/useInputState';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import I18n from '../../lib/i18n';
+import { RECAPTCHA_PUBLIC_KEY } from '../../lib/wca-data.js.erb';
+import UserData from './SubForms/UserData';
+import Loading from '../Requests/Loading';
+import Wct from './SubForms/Wct';
+import Competition from './SubForms/Competition';
+
+const CONTACT_TYPES = [
+  'competition',
+  'competitions_in_general',
+  'results_team',
+  'wca_id_or_profile',
+  'media',
+  'software',
+  'different',
+];
+
+const CONTACT_TYPES_MAP = CONTACT_TYPES.reduce((accumulator, contactType) => {
+  accumulator[contactType] = contactType;
+  return accumulator;
+}, {});
+
+export default function ContactForm({ userDetails }) {
+  const initialFormValues = useMemo(() => ({
+    userData: {
+      name: userDetails?.user?.name || '',
+      email: userDetails?.user?.email || '',
+    },
+  }), [userDetails]);
+  const [formValues, setFormValues] = useInputState(initialFormValues);
```

Comment:
Hm, reading your other comments makes me think that you're stuffing too much into the `formValues` state here.
I believe you should have an independent state for "which toggle is selected" and then another state for the "sub form values", i.e. all relevant information from the form.

Another alternative would be to introduce a reducer for state management. I have a slight preference towards the first option, but ultimately I leave it up to you to decide which is better suited here.